Userland/Games/BrickGame/Tetris.h

```diff
@@ -0,0 +1,343 @@
+
+        // Shape: S
+        { 0b0000'0011'0110'0000, 0b0100'0110'0010'0000, 0b0000'0011'0110'0000,
+            0b0100'0110'0010'0000 },
+
+        // Shape: T
+        { 0b0000'0111'0010'0000, 0b0001'0011'0001'0000, 0b0000'0010'0111'0000,
+            0b0100'0110'0100'0000 },
+
+        // Shape: Z
+        { 0b0000'0110'0011'0000, 0b0001'0011'0010'0000, 0b0000'0110'0011'0000,
+            0b0001'0011'0010'0000 }
+    };
+
+    Position m_pos {};
+    u8 m_rotation {};
+    u8 m_shape {};
+
+    [[nodiscard]] Row block_row(size_t row) const
+    {
+        auto const block_data = s_patterns[m_shape][m_rotation];
+        auto const x = 32 - m_pos.x() - shape_size;
+        switch (row) {
+        case 0:
+            return Row((block_data >> 12) & 0xf) << x;
+        case 1:
+            return Row((block_data >> 8) & 0xf) << x;
+        case 2:
+            return Row((block_data >> 4) & 0xf) << x;
+        case 3:
+            return Row(block_data & 0xf) << x;
+        default:
+            return Row {};
+        }
+    }
+
+public:
+    Position position() const { return m_pos; }
+
+    Block() = default;
+    Block(Block const&) = default;
+    Block& operator=(Block const&) = default;
+
+    Block& rotate_left()
+    {
+        m_rotation = m_rotation == 0 ? num_rotations - 1 : m_rotation - 1;
+        return *this;
+    }
+
+    Block& rotate_right()
+    {
+        m_rotation = m_rotation == num_rotations - 1 ? 0 : m_rotation + 1;
+        return *this;
+    }
+
+    Block& move_left()
+    {
+        m_pos = m_pos.moved_left(1);
+        return *this;
+    }
+
+    Block& move_right()
+    {
+        m_pos = m_pos.moved_right(1);
+        return *this;
+    }
+
+    Block& move_down()
+    {
+        m_pos = m_pos.moved_down(1);
+        return *this;
+    }
+
+    Block& move_to(Position pos)
+    {
+        m_pos = pos;
+        return *this;
+    }
+
+    Block& random_shape()
+    {
+        m_shape = AK::get_random_uniform(num_shapes);
+        m_rotation = 0;
+        m_pos = { 6, 0 };
+        return *this;
+    }
+
+    [[nodiscard]] bool operator[](Position pos) const
+    {
+        return pos.y() >= m_pos.y() && (block_row(pos.y() - m_pos.y()) & (1 << (31 - pos.x()))) != 0;
+    }
+
+    [[nodiscard]] bool has_collision(Well const& well) const
+    {
+        for (size_t start_row = 0; start_row < shape_size; ++start_row) {
+            auto const row_index = start_row + m_pos.y();
+            if (row_index >= Well::num_rows() || (well[row_index] & block_row(start_row)) != 0)
+                return true;
+        }
+        return false;
+    }
+
+    bool place_into(Well& well)
+    {
+        for (size_t i = 0; i < shape_size; ++i) {
+            well[m_pos.y() + i] |= block_row(i);
+        }
+        return true;
+    }
+};
+
+class Tetris final {
+    Well m_well;
+    Block m_block;
+    unsigned m_level {};
+    unsigned m_score {};
+    bool m_is_game_over {};
+    AK::Time m_last_update;
```

Comment:
Member variables should be declared as `private`
Additionally, private variables should be laid out after public ones.